data_structures/binary_tree/wavelet_tree.py

```diff
@@ -34,6 +34,11 @@ def build_tree(arr: List[int]) -> Node:
     """
     Builds the tree for arr and returns the root
     of the constructed tree
+
+    >>> arr = [2,1,4,5,6,8,9,1,2,6,7,4,2,6,5,3,2,7]
+    >>> root = build_tree(arr)
+    >>> root
```

Comment:
Create a `test_array = [2, 1, 4, 5, 6, 8, 9, 1, 2, 6, 7, 4, 2, 6, 5, 3, 2, 7]` variable at global scope _near the top of the file_ that we can use repeatedly to shorten our doctest...

Should `test_array` have negative numbers?  zero? floats?
```suggestion
    >>> build_tree(test_array)
```

Reply:
test_array can contain any number(int or float) irrespective of the value range, but in one of the functions I'm returning -1 to raise as an error.